rotkehlchen/db/addressbook.py

```diff
@@ -64,65 +94,87 @@ def add_addressbook_entries(
                     )
                 # Handling both private db (pysqlcipher) and global db (raw sqlite3)
                 except (dbapi2.IntegrityError, sqlite3.IntegrityError) as e:  # pylint: disable=no-member  # noqa: E501
                     raise InputError(
-                        f'Addressbook entry with address "{entry.address}" and name "{entry.name}"'
-                        f' already exists in the address book. Identifier must be unique.',
+                        f'Addressbook entry with address "{entry.address}", name "{entry.name}" '
+                        f'and chain {entry.blockchain} already exists in the address book. '
+                        f'Identifier must be unique.',
                     ) from e
 
     def update_addressbook_entries(
             self,
             book_type: AddressbookType,
             entries: list[AddressbookEntry],
     ) -> None:
+        """
+        Updates names of addressbook entries.
+        If blockchain is None then update all entries with the specified address.
+        """
         with self.write_ctx(book_type) as write_cursor:
             for entry in entries:
-                write_cursor.execute(
-                    'UPDATE address_book SET name = ? WHERE address = ?',
-                    (entry.name, entry.address),
-                )
+                query = 'UPDATE address_book SET name = ? WHERE address = ? AND blockchain = ?'
+                bindings = (entry.name, entry.address, entry.blockchain.value if entry.blockchain else None)  # noqa: E501
+                write_cursor.execute(query, bindings)
                 if write_cursor.rowcount == 0:
                     raise InputError(
-                        f'Addressbook entry with address "{entry.address}" and name "{entry.name}"'
-                        f' doesn\'t exist in the address book. So it cannot be modified.',
+                        f'Addressbook entry with address "{entry.address}" and name '
+                        f'"{entry.name}" and blockchain {entry.blockchain} doesn\'t exist '
+                        f'in the address book. So it cannot be modified.',
                     )
 
     def delete_addressbook_entries(
             self,
             book_type: AddressbookType,
-            addresses: list[ChecksumEvmAddress],
+            addresses: list[OptionalChainAddress],
     ) -> None:
         with self.write_ctx(book_type) as write_cursor:
-            for address in addresses:
-                write_cursor.execute('DELETE FROM address_book WHERE address = ?', (address,))
+            for address, blockchain in addresses:
+                query = 'DELETE FROM address_book WHERE address = ?'
+                bindings: list[str] = [address]
+                if blockchain is not None:
+                    query += ' AND blockchain = ?'
+                    bindings.append(blockchain.value)
+                write_cursor.execute(query, bindings)
```

Comment:
The reason is to properly format the error when returned to the frontend. Otherwise you don't know which one failed

Reply:
this is not a good enough reason. You don't need to know which specific name failed. Unless it's an explicit requirement for some reason I don't get understand, please optimize for the least number of DB writes.

Reply:
Found a case where this is needed. If we delete by address only it can happen that the address has more than one entry (different chains) and this might result in a error detecting addresses that couldn't be deleted because they dont' exist in the db